3rdparty/stout/include/stout/decompress.hpp

```diff
@@ -0,0 +1,159 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef __STOUT_DECOMPRESS_HPP__
+#define __STOUT_DECOMPRESS_HPP__
+
+#include <archive.h>
+#include <archive_entry.h>
+
+#include <glog/logging.h>
+
+#include <stout/nothing.hpp>
+#include <stout/path.hpp>
+#include <stout/try.hpp>
+
+namespace decompress {
+
+enum Format
+{
+  // These are a subset of formats supported by libarchive, but this
+  // is likely all that Mesos cares about.
+  FORMAT_ANY = 0,
+  FORMAT_BZIP2,
+  FORMAT_COMPRESS,
+  FORMAT_CPIO,
+  FORMAT_GZIP,
+  FORMAT_TAR,
+  FORMAT_ZIP
+};
+
+enum Flags
```

Comment:
Probably over-abstracted; we can global replace these later if we for some reason we change from libarchive.

Reply:
Okay, fixed.